neo4j/_async_compat/network/_bolt_socket.py

```diff
@@ -90,11 +91,18 @@ async def _wait_for_io(self, io_fut):
         except asyncio.TimeoutError as e:
             raise to_raise("timed out") from e
 
-    def set_deadline(self, s_from_now):
-        if s_from_now is None:
+    def get_deadline(self):
+        return self._deadline
+
+    def set_deadline(self, deadline):
```

Comment:
this logic of validating an actual value is duplicated(line 415). maybe define it as a function and use it in both places of this file.

Reply:
Good point. :broom: addressed.